src/LinuxTracing/LeafFunctionCallManager.cpp

```diff
@@ -90,10 +84,13 @@ Callstack::CallstackType LeafFunctionCallManager::PatchCallerOfLeafFunction(
     return Callstack::kStackTopDwarfUnwindingError;
   }
 
-  // In case of an intact frame pointer, the region from $rsp to $rbp will only include the locals
-  // of the current frame, and NOT the return address. Thus, unwinding will only report one frame
-  // (the instruction pointer) and we know that the original callchain is already correct.
   if (libunwindstack_callstack.size() == 1) {
+    if (stack_size > stack_dump_size_) {
+      return Callstack::kStackTopForDwarfUnwindingTooSmall;
+    }
```

Comment:
Even if `libunwindstack_result` contains no error? Can that even happen?

Reply:
But I guess you are right. It also does not hurt to check for the libunwindstack result.

Reply:
Makes sense. I was wondering in what cases this could happen. I think it's good to handle it then, if anything on a conceptual level, so that a non-error is turned into an error without explanation.